shopfloor_mobile/static/wms/src/scenario/form.js

```diff
@@ -0,0 +1,107 @@
+/**
+ * Copyright 2020 Camptocamp SA (http://www.camptocamp.com)
+ * @author Simone Orsi <simahawk@gmail.com>
+ * License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+ */
+
+import {process_registry} from "../services/process_registry.js";
+
+const Form = {
+    template: `
+        <Screen :screen_info="screen_info">
+            <component
+                :is="form_component_name()"
+                :record="record"
+                :form="form"
+                v-on:change="on_change"
+                />
+            <div class="button-list button-vertical-list full" v-if="!_.isEmpty(record)">
+                <v-row align="center">
+                    <v-col class="text-center">
+                        <btn-back />
+                    </v-col>
+                </v-row>
+            </div>
+        </Screen>
+    `,
+    data: function() {
+        return {
+            usage: "",
+            form_name: "",
+            record: {},
+            form: {},
+            user_message: {},
+        };
+    },
+    mounted: function() {
+        this.usage = this.$route.params.form_name;
+        const odoo_params = {
+            usage: this.usage,
+            profile_id: this.$root.profile.id,
+        };
+        this.odoo = this.$root.getOdoo(odoo_params);
+        if (this.$route.params.record_id) {
+            this._getFormData(this.$route.params.record_id);
+        }
+    },
+    methods: {
+        on_change: function(form_data) {
+            if (!form_data.changed) {
+                return;
+            }
+            const self = this;
+            let form_values = form_data.values;
+            form_values["_id"] = this.record.id;
+            this.odoo.call("update", form_values).then(function(result) {
+                self._load_form_data(result);
+                setTimeout(function() {
+                    self.$root.trigger("go_back");
+                    self.$router.back();
+                }, 1500);
```

Comment:
is there a rationale for this duration?

Reply:
bah, it's just a trick to get the success message displayed and then redirect to where you were before.
It should be improved somehow by passing the message to parent component.
I'll add a TODO.

Reply:
done